src/types.rs

```diff
@@ -241,6 +231,29 @@ macro_rules! key_methods {
                 Ok(ret)
             }
         }
+
+        #[cfg(feature = "serde")]
```

Comment:
I don't think its an improvement to move out of proc derived serde implementation to manual/customized implementations. Why would we want that?